translator/trace/zipkin/protospan_to_zipkinv1_test.go

```diff
@@ -119,3 +135,153 @@ func TestZipkinEndpointFromNode(t *testing.T) {
 		})
 	}
 }
+
+func TestOCSpanProtoToZipkin_endToEnd(t *testing.T) {
+	// The goal of this test is to ensure that each
+	// tracepb.Span is transformed to its *trace.SpanData correctly!
+
+	endTime := time.Now().Round(time.Second)
+	startTime := endTime.Add(-90 * time.Second)
+
+	protoSpan := &tracepb.Span{
+		TraceId:      []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F},
+		SpanId:       []byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8},
+		ParentSpanId: []byte{0xEF, 0xEE, 0xED, 0xEC, 0xEB, 0xEA, 0xE9, 0xE8},
+		Name:         &tracepb.TruncatableString{Value: "End-To-End Here"},
+		Kind:         tracepb.Span_SERVER,
+		StartTime:    internal.TimeToTimestamp(startTime),
+		EndTime:      internal.TimeToTimestamp(endTime),
+		Status: &tracepb.Status{
+			Code:    13,
+			Message: "This is not a drill!",
+		},
+		SameProcessAsParentSpan: &wrappers.BoolValue{Value: false},
+		TimeEvents: &tracepb.Span_TimeEvents{
+			TimeEvent: []*tracepb.Span_TimeEvent{
+				{
+					Time: internal.TimeToTimestamp(startTime),
+					Value: &tracepb.Span_TimeEvent_MessageEvent_{
+						MessageEvent: &tracepb.Span_TimeEvent_MessageEvent{
+							Type: tracepb.Span_TimeEvent_MessageEvent_RECEIVED, UncompressedSize: 1024, CompressedSize: 512,
+						},
+					},
+				},
+				{
+					Time: internal.TimeToTimestamp(endTime),
+					Value: &tracepb.Span_TimeEvent_MessageEvent_{
+						MessageEvent: &tracepb.Span_TimeEvent_MessageEvent{
+							Type: tracepb.Span_TimeEvent_MessageEvent_SENT, UncompressedSize: 1024, CompressedSize: 1000,
+						},
+					},
+				},
+				{
+					Time: internal.TimeToTimestamp(endTime),
+					Value: &tracepb.Span_TimeEvent_Annotation_{
+						Annotation: &tracepb.Span_TimeEvent_Annotation{
+							Description: &tracepb.TruncatableString{Value: "Description"},
+						},
+					},
+				},
+			},
+		},
+		Links: &tracepb.Span_Links{
+			Link: []*tracepb.Span_Link{
+				{
+					TraceId: []byte{0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF},
+					SpanId:  []byte{0xB0, 0xB1, 0xB2, 0xB3, 0xB4, 0xB5, 0xB6, 0xB7},
+					Type:    tracepb.Span_Link_PARENT_LINKED_SPAN,
+				},
+				{
+					TraceId: []byte{0xE0, 0xE1, 0xE2, 0xE3, 0xE4, 0xE5, 0xE6, 0xE7, 0xE8, 0xE9, 0xEA, 0xEB, 0xEC, 0xED, 0xEE, 0xEF},
+					SpanId:  []byte{0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0xD5, 0xD6, 0xD7},
+					Type:    tracepb.Span_Link_CHILD_LINKED_SPAN,
+				},
+			},
+		},
+		Tracestate: &tracepb.Span_Tracestate{
+			Entries: []*tracepb.Span_Tracestate_Entry{
+				{Key: "foo", Value: "bar"},
+				{Key: "a", Value: "b"},
+			},
+		},
+		Attributes: &tracepb.Span_Attributes{
+			AttributeMap: map[string]*tracepb.AttributeValue{
```

Comment:
Could be good to also have a test that covers "span.kind"

Reply:
Already some tests in TestSpanKindTranslation (zipkinv1_to_protospan_test.go)